### DISCUSSION TITLE
Clone()

### DISCUSSION THREAD

**Discussion by sergolius** [uptrace/bun#342] [Q&A] [answered]
Would you implement `Clone()` method for queries?
Similar to `go-pg` `&Query{}.Clone()`

**Comment by vmihailenco** [comment, marked as answer]
It was a constant source of bugs in go-pg and in Bun we have even more queries to support, but I would accept a PR adding Clone method.

The recommended way is to extract query building into a method and call it whenever you need a duplicate.
